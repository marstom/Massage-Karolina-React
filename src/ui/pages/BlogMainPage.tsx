import React from "react";
import styled from "styled-components";
import { BlogMinifiedEntry } from "../atoms/blog_minified_entry";
import { Content } from "ui/atoms/Content";

type Props = {};

export const BlogMainPage = (props: Props) => {
  return (
    <Content>
      <BlogMinifiedEntry title={"Title"} content={"To jest opis"} />
      <BlogMinifiedEntry
        title={"Inny post"}
        content={"To jest opi asdf asdfseffaes eaf "}
      />
      <BlogMinifiedEntry
        title={"Inny post"}
        content={"To jest opi asdf asdfseffaes eaf "}
      />
    </Content>
  );
};
