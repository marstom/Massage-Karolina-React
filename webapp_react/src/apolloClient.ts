import { ApolloClient, InMemoryCache } from "@apollo/client";

export const client = new ApolloClient({
  uri: `${process.env.REACT_APP_BASE_BACKEND_URL}/graphql`,
  cache: new InMemoryCache(),
});
