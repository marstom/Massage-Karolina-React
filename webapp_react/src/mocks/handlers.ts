// src/mocks/handlers.js
import { graphql } from 'msw'
import {postResponse, postsResponse} from "./responses/postsResponse";

export const handlers = [

    graphql.query('Posts', (req, res, ctx) => {
        return res(
            ctx.data(postsResponse),
        )
    }),
    graphql.query('Post', (req, res, ctx) => {
        const { id } = req.variables
        return res(
            ctx.data(postResponse(id)),
        )
    }),
]

