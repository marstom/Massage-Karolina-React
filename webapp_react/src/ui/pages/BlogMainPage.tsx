import React from "react";
import { BlogMinifiedEntry } from "../atoms/BlogMinifiedEntry";
import { Content } from "ui/atoms/Content";
import { gql, useQuery } from "@apollo/client";
import { Posts } from "ui/types/blogPosts";
import styled from "styled-components";
import { BASE_URL } from "../../consts";

const postsQuery = gql`
  query Posts {
    posts(sort: "createdAt:desc") {
      data {
        id
        attributes {
          title
          body
          miniImage {
            data {
              attributes {
                url
                caption
              }
            }
          }
        }
      }
    }
  }
`;

const BlogFlexContent = styled(Content)`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin-left: auto;
  justify-content: center;
  width: 100%;
`;

export const BlogMainPage: React.FC<{}> = () => {
  const { loading, error, data } = useQuery<Posts>(postsQuery);

  if (loading) return <p>Loading...</p>;
  if (error) return <p>Error :( mmm</p>;
  return (
    <BlogFlexContent>
      {data!.posts.data.map((post) => (
        <BlogMinifiedEntry
          imageUrl={BASE_URL + post.attributes?.miniImage?.data?.attributes.url}
          key={post.id}
          id={post.id}
          title={post.attributes.title}
          content={post.attributes.body}
        />
      ))}
    </BlogFlexContent>
  );
};
