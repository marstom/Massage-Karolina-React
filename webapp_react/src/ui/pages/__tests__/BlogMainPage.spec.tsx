import React from "react";
import { server } from "../../../mocks/server";
import { BlogMainPage } from "../BlogMainPage";
import { render, waitFor } from "@testing-library/react";
import { ApolloProvider } from "@apollo/client";
import { client } from "../../../apolloClient";
import { BrowserRouter } from "react-router-dom";

beforeAll(() => server.listen());

// Reset any request handlers that we may add during the tests,
// so they don't affect other tests.
afterEach(() => server.resetHandlers());

// Clean up after the tests are finished.

afterAll(() => server.close());

describe("This is blog tests", () => {
  it("should render all block posts from api mock properly and should match to snapshot", async () => {
    const { container, findByText } = render(
      <ApolloProvider client={client}>
        <BrowserRouter>
          <BlogMainPage />
        </BrowserRouter>
      </ApolloProvider>
    );

    const firstPostTitle = await findByText("Co to jest masaż tantryczny?");
    expect(firstPostTitle).toMatchInlineSnapshot(`
      <h1>
        Co to jest masaż tantryczny?
      </h1>
    `);
    expect(container).toMatchSnapshot();
  });
});
