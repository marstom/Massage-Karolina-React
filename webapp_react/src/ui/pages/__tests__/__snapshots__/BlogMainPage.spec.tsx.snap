// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This is blog tests should render all block posts from api mock properly and should match to snapshot 2`] = `
<div>
  <div
    class="sc-hKgJyM jraeqF"
  >
    <div
      class="sc-gsTChe igtfLU"
    >
      <h2>
        Tantra
      </h2>
      <div>
        <h1>
          Co to jest masaż tantryczny?
        </h1>
        <p>
          Masaż tantryczny jest rodzajem pracy z dotykiem, której idea zaczerpnięta została ze starożytnej tradycji hinduistycznej i buddyjskiej.
        </p>
      </div>
      <a
        class="sc-bdfCkl sc-dlfmnT gASFcl leRbyo"
        href="/blog/1"
      >
        Czytaj dalej...
      </a>
    </div>
    <div
      class="sc-gsTChe igtfLU"
    >
      <h2>
        Lomi-lomi
      </h2>
      <div>
        (api mock)Masaż Lomi Lomi wywodzi się z Hawajów. Wieki temu wykonywano go w świątyniach, gdy osoba poddająca się zabiegowi miała podjąć jakąś ważną decyzje życiową. Ten egzotyczny rytuał poprzez dotyk głęboko oczyszcza duchowo i fizycznie, a także przywraca równowagę.
      </div>
      <a
        class="sc-bdfCkl sc-dlfmnT gASFcl leRbyo"
        href="/blog/2"
      >
        Czytaj dalej...
      </a>
    </div>
    <div
      class="sc-gsTChe igtfLU"
    >
      <h2>
        Level 20
      </h2>
      <div>
        <h1>
          To jest tytuł
        </h1>
        <p>
          There is a setuid binary in the homedirectory that does the following:** it makes a connection to localhost on the port you specify as a commandline argument.
        </p>
      </div>
      <a
        class="sc-bdfCkl sc-dlfmnT gASFcl leRbyo"
        href="/blog/3"
      >
        Czytaj dalej...
      </a>
    </div>
    <div
      class="sc-gsTChe igtfLU"
    >
      <h2>
        Naj
      </h2>
      <div>
        <p>
          Najnowszy post
        </p>
      </div>
      <a
        class="sc-bdfCkl sc-dlfmnT gASFcl leRbyo"
        href="/blog/4"
      >
        Czytaj dalej...
      </a>
    </div>
  </div>
</div>
`;
